cns/deviceplugin/pluginmanager_test.go

```diff
@@ -0,0 +1,303 @@
+	req := &v1beta1.AllocateRequest{
+		ContainerRequests: []*v1beta1.ContainerAllocateRequest{
+			{
+				DevicesIDs: []string{"device-0", "device-1"},
+			},
+		},
+	}
+	allocateResp := getAllocateResponse(t, vnetPluginEndpoint, req)
+
+	if len(allocateResp.ContainerResponses[0].Envs) != len(req.ContainerRequests[0].DevicesIDs) {
+		t.Fatalf("expected allocations %v but received allocations %v", len(req.ContainerRequests[0].DevicesIDs), len(allocateResp.ContainerResponses[0].Envs))
+	}
+
+	// call getDevicePluginOptions method
+	_, err = getDevicePluginOptionsResponse(vnetPluginEndpoint)
+	if err != nil {
+		t.Fatalf("error calling getDevicePluginOptions: %v", err)
+	}
+
+	// call getPreferredAllocation method
+	_, err = getPreferredAllocationResponse(vnetPluginEndpoint)
+	if err != nil {
+		t.Fatalf("error calling getPreferredAllocation: %v", err)
+	}
+
+	// call preStartContainer method
+	_, err = getPreStartContainerResponse(vnetPluginEndpoint)
+	if err != nil {
+		t.Fatalf("error calling PreStartContainer: %v", err)
+	}
+
+	// shut down the plugin manager and fake kubelet
+	cancel()
+
+	// ensure the plugin manager didn't report an error
+	if err := <-errChan; err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// ensure the fake kubelet didn't report an error
+	if err := <-kubeletErrChan; err != nil {
+		t.Fatalf("unexpected error from fake kubelet: %v", err)
+	}
+}
+
+type fakeKubelet struct {
+	vnetPluginRegisterChan chan string
+	ibPluginRegisterChan   chan string
+	pluginPrefix           string
+}
+
+func (f *fakeKubelet) Register(_ context.Context, req *v1beta1.RegisterRequest) (*v1beta1.Empty, error) {
+	switch req.ResourceName {
+	case string(v1alpha1.DeviceTypeVnetNIC):
+		f.vnetPluginRegisterChan <- path.Join(f.pluginPrefix, req.Endpoint)
+	case string(v1alpha1.DeviceTypeInfiniBandNIC):
+		f.ibPluginRegisterChan <- path.Join(f.pluginPrefix, req.Endpoint)
+	}
+	return &v1beta1.Empty{}, nil
+}
+
+func runFakeKubelet(ctx context.Context, address string, vnetPluginRegisterChan, ibPluginRegisterChan chan string, pluginPrefix string) error {
+	if err := os.Remove(address); err != nil && !os.IsNotExist(err) {
+		return errors.Wrap(err, "error cleaning up previous kubelet socket")
+	}
+
+	k := &fakeKubelet{
+		vnetPluginRegisterChan: vnetPluginRegisterChan,
+		ibPluginRegisterChan:   ibPluginRegisterChan,
+		pluginPrefix:           pluginPrefix,
+	}
+	grpcServer := grpc.NewServer()
+	v1beta1.RegisterRegistrationServer(grpcServer, k)
+
+	l, err := net.Listen("unix", address)
+	if err != nil {
+		return errors.Wrap(err, "error from fake kubelet listening on socket")
+	}
+	errChan := make(chan error, 2)
+	go func() {
+		errChan <- grpcServer.Serve(l)
+	}()
+	defer grpcServer.Stop()
+
+	select {
+	case err := <-errChan:
+		return errors.Wrap(err, "error running fake kubelet grpc server")
+	case <-ctx.Done():
+	}
+	return nil
+}
+
+//nolint:staticcheck // TODO: Move to grpc.NewClient method
```

Comment:
could we move these nolint comments directly to the line we're trying to skip? having it here skips staticcheck for the whole function which we should try to avoid

Reply:
moved